infra/src/main/kotlin/fi/fta/geoviite/infra/linking/PublicationService.kt

```diff
@@ -142,6 +142,73 @@ class PublishService @Autowired constructor(
         }
     }
 
+    @Transactional(readOnly = true)
+    fun getRevertRequestDependencies(publishRequest: PublishRequest): PublishRequest {
+        val newTrackNumbers = publishRequest.referenceLines.mapNotNull { id -> referenceLineService.get(DRAFT, id) }
+            .map { rl -> rl.trackNumberId }
+            .filter { id -> trackNumberService.get(DRAFT, id) != null }
+
+        val newReferenceLines = publishRequest.trackNumbers.mapNotNull { id -> referenceLineDao.fetchVersion(DRAFT, id) }
+            .map { version -> version.id }
+        val allTrackNumbers = publishRequest.trackNumbers.toSet() + newTrackNumbers.toSet()
+        val allReferenceLines = publishRequest.referenceLines.toSet() + newReferenceLines.toSet()
+        val locationTracks = publishRequest.locationTracks.toSet()
+        val switches = publishRequest.switches.toSet()
+        val (allLocationTracks, allSwitches) = getRevertRequestLocationTrackAndSwitchDependenciesTransitively(
+            locationTracks,
+            locationTracks,
+            switches,
+            switches
+        )
+
+        return PublishRequest (
+            trackNumbers = allTrackNumbers.toList(),
+            referenceLines = allReferenceLines.toList(),
+            locationTracks = allLocationTracks.toList(),
+            switches = allSwitches.toList(),
+            kmPosts = publishRequest.kmPosts
+        )
+    }
+
+    private fun getRevertRequestLocationTrackAndSwitchDependenciesTransitively(
+        allLocationTracks: Set<IntId<LocationTrack>>,
+        lastLevelLocationTracks: Set<IntId<LocationTrack>>,
+        allSwitches: Set<IntId<TrackLayoutSwitch>>,
```

Comment:
Ottaen huomioon että näitä `all`-nimettyjä parametreja käytetään subtractiin, mulle ei jäänyt kauhean hyvä kuva siitä mitä nämä oikeasti sisältävät. Nimenä nuo viittaisivat siihen, että noissa olisi sijaintiraiteiden ja vaihteiden koko maailmat. Kun jostain joukosta vähennetään koko maailma, niin tuloksena pitäisi aina olla tyhjä joukko, mutta näin ei kuitenkaan liene.